backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/rules/RulesData.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.impl.rules;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import lombok.*;
+import lombok.extern.slf4j.Slf4j;
+import org.hyperledger.aries.api.connection.ConnectionRecord;
+import org.hyperledger.aries.api.connection.ConnectionState;
+import org.hyperledger.aries.api.connection.ConnectionTheirRole;
+import org.hyperledger.bpa.model.ActiveRules;
+
+import java.util.UUID;
+
+@Slf4j
+@Data
+@Builder
+public class RulesData {
+
+    private UUID ruleId;
+    private Trigger trigger;
+    private Action action;
+
+    @JsonTypeInfo(
+            use = JsonTypeInfo.Id.NAME,
+            property = "type")
+    @JsonSubTypes({
+            @JsonSubTypes.Type(value = Trigger.ConnectionTrigger.class, name = Trigger.CONNECTION_TRIGGER_NAME),
+            @JsonSubTypes.Type(value = Trigger.ProofReceivedTrigger.class, name = Trigger.PROOF_RECEIVED_TRIGGER_NAME)
+    })
+    @NoArgsConstructor
+    public abstract static class Trigger {
+
+        public static final String CONNECTION_TRIGGER_NAME = "connection";
+        public static final String PROOF_RECEIVED_TRIGGER_NAME = "proof_received";
+
+        abstract boolean apply(EventContext ctx);
+
+        @SuppressWarnings("unused")
+        private String type;
```

Comment:
What is the purpose of this field? It's not used via accessors and builders. The database stores Triggers as jsonb.

How about pulling this field into the subclasses, if there is a purpose for it. And then change Trigger to an interface, which is the also a functional one. Or does that create serialization issues?